ui/src/main/java/com/dat3m/ui/Dat3M.java

```diff
@@ -4,111 +4,161 @@
-    	EditorListener listener = new EditorListener(optionsPane.getConsolePane());
-    	editorsPane.getEditor(EditorCode.PROGRAM).getEditorPane().addKeyListener(listener);
-    	editorsPane.getEditor(EditorCode.TARGET_MM).getEditorPane().addKeyListener(listener);
-
-		pack();
-	}
-
-	public static void main(String[] args) {
-		EventQueue.invokeLater(() -> {
-			Dat3M app = new Dat3M();
-			app.setVisible(true);
-		});
-	}
-
-	@Override
-	public void actionPerformed(ActionEvent event) {
-	    String command = event.getActionCommand();
-	    if(ControlCode.TEST.actionCommand().equals(command)){
+    private final OptionsPane optionsPane = new OptionsPane();
+    private final EditorsPane editorsPane = new EditorsPane();
+
+    private ReachabilityResult testResult;
+
+    private Dat3M() {
+        getDefaults().put("SplitPane.border", createEmptyBorder());
+
+        setTitle("Dat3M");
+        setExtendedState(JFrame.MAXIMIZED_BOTH);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+//      setIconImage(IconHelper.getIcon(IconCode.DAT3M).getImage());
+
+        JMenuBar menuBar = new JMenuBar();
+        JMenu fileMenu = new JMenu("File");
+        fileMenu.add(editorsPane.getMenuImporter());
+        fileMenu.add(editorsPane.getMenuExporter());
+        menuBar.add(fileMenu);
+        setJMenuBar(menuBar);
+
+        JSplitPane mainPane = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT, optionsPane, editorsPane.getMainPane());
+        mainPane.setDividerSize(2);
+        add(mainPane);
+
+        // Start listening to button events
+        optionsPane.getTestButton().addActionListener(this);
+
+        // optionsPane needs to listen to editor to clean the console
+        editorsPane.getEditor(EditorCode.PROGRAM).addActionListener(optionsPane);
+        editorsPane.getEditor(EditorCode.TARGET_MM).addActionListener(optionsPane);
+
+        // The console shall be cleaned every time the program or MM is modified from the editor
+        EditorListener listener = new EditorListener(optionsPane.getConsolePane());
+        editorsPane.getEditor(EditorCode.PROGRAM).getEditorPane().addKeyListener(listener);
+        editorsPane.getEditor(EditorCode.TARGET_MM).getEditorPane().addKeyListener(listener);
+
+        pack();
+    }
+
+    public static void main(String[] args) {
+        EventQueue.invokeLater(() -> {
+            Dat3M app = new Dat3M();
+            app.setVisible(true);
+        });
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent event) {
+        String command = event.getActionCommand();
+        if (ControlCode.TEST.actionCommand().equals(command)) {
             runTest();
-            if(testResult != null){
+            if (testResult != null) {
                 optionsPane.getConsolePane().setText(testResult.getVerdict());
+                if (optionsPane.getOptions().showWitness() && testResult.hasWitness()) {
```

Comment:
I find annoying that if the "generate witness" button is marked (I would expect user to click this once and keep it marked rather than selecting/unselecting each time they run the verification), we generate a witness for the UNKNOWN result.

Rather than checking if there is a witness, IMO we should check if the result is FAIL.

I don't think the argument "we also generate a witness if you pass the option --witness=png to the console and the result is UNKNOWN" is valid for the simple reason that it creates a file that does not disturb the user. Contrary to this, when running on the CI a new windows will pop up.

Reply:
I can change that. But your original argument was that it was not aligned with the console behaviour although it was.

Reply:
My bad based on what I actually wrote. What I meant was "it is different in the sense in the UI something happened that annoys the user whole in the console it does not"